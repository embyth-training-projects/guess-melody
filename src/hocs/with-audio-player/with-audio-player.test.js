import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import withAudioPlayer from "./with-audio-player";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const MockComponentWrapped = withAudioPlayer(MockComponent);

it(`withAudioPlayer is rendered correctly`, () => {
  const tree = renderer
    .create(<MockComponentWrapped
      isPlaying={false}
      onPlayButtonClick={() => {}}
      src={``}
    />, {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
