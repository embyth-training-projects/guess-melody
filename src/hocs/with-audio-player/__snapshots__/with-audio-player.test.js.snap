// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAudioPlayer is rendered correctly 1`] = `
<div>
  <audio />
</div>
`;
